src/Fields/background_fields.jl

```diff
@@ -60,6 +60,8 @@ func(x, y, z, t, parameters)
 """
 BackgroundField(func; parameters=nothing) = BackgroundField(func, parameters)
 
+regularize_background_field(LX, LY, LZ, c::Number, grid, clock) = ConstantField(c)
```

Comment:
```suggestion
regularize_background_field(LX, LY, LZ, bg::BackgroundField{<:Number}, grid, clock) = ConstantField(bg.func)
```

My b, this is what it should look like

I guess we assumed the background fields would be functions and called that property `func`...

I'd chalk up that cleanup to also resolving #3586 not worth tackling here